core/engine/src/value/mod.rs

```diff
@@ -225,7 +285,11 @@ impl JsValue {
     #[inline]
     #[must_use]
     pub fn is_regexp(&self) -> bool {
-        matches!(self, Self::Object(obj) if obj.is::<RegExp>())
+        if let InnerValue::Object(obj) = &self.inner {
```

Comment:
nit (non-blocking): Here and elsewhere, there seems to be different styles used on `is_x` methods.

For instance, couldn't this just be `matches!(self, InnerValue::Object(obj)) if obj.is::<RegExp>`. This is mostly just a stylistic nit with just consistency across the methods in mind.

Reply:
`matches!()` won't work here and elsewhere in the next step anyway as we'll need to unbox the NaN value. I'll merge as is. Thanks!